File: blog/src/main/kotlin/com/liux/blog/bean/po/Fragment.kt
package com.liux.blog.bean.po

import com.liux.blog.bean.NoArgsConstructor
import java.io.Serializable
import java.util.*

// 文章数据
@NoArgsConstructor
data class Fragment(
    // ID
    var id: Int? = null,
    // KEY
    var key: String? = null,
    // 内容
    var content: String? = null,
    // 作者ID
    var authorId: Int? = null,
    // 状态
    var status: Int? = null,
    // 创建时间
    var createTime: Date? = null,
    // 更新时间
    var updateTime: Date? = null,

    /* 作者 */
    var author: User? = null,
) : Serializable

File: blog/bean/api_vo/fragment.go
package api_vo

import (
	"blog/bean/po"
	"time"
)

type FragmentVO struct {
	Id         int        `json:"id"`
	Key        string     `json:"key"`
	Content    string     `json:"content"`
	AuthorId   int        `json:"authorId"`
	Status     int        `json:"status"`
	CreateTime time.Time  `json:"createTime"`
	UpdateTime *time.Time `json:"updateTime"`
}

func (v *FragmentVO) From(fragment po.Fragment) {
	v.Id = fragment.Id
	v.Key = fragment.Key
	v.Content = fragment.Content
	v.AuthorId = fragment.AuthorId
	v.Status = fragment.Status
	v.CreateTime = fragment.CreateTime
	v.UpdateTime = fragment.UpdateTime
}
